homeassistant/components/redgtech/coordinator.py

```diff
@@ -0,0 +1,99 @@
+"""Coordinator for Redgtech integration."""
+
+from __future__ import annotations
+
+import logging
+from datetime import timedelta
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+from homeassistant.core import HomeAssistant
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import STATE_ON, STATE_OFF, CONF_ACCESS_TOKEN, CONF_EMAIL, CONF_PASSWORD
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+from homeassistant.exceptions import HomeAssistantError, ConfigEntryError, ConfigEntryNotReady
+from redgtech_api.api import RedgtechAPI, RedgtechAuthError, RedgtechConnectionError
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+@dataclass
+class RedgtechDevice:
+    """Representation of a Redgtech device."""
+    id: str
```

Comment:
```suggestion
    unique_id: str
```
`id` is a built in function from python, let's not override that